### PR TITLE
Add Soak tests for Kafka connect source [HZ-1814]

### PR DESCRIPTION
**Pull request by orcunc**

### CODE REVIEW

File: kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Properties;
+import java.util.concurrent.CompletionException;
+
+public class KafkaConnectSourceTest extends AbstractSoakTest {
+
+    private static final String CONNECTOR_URL = "https://repository.hazelcast.com/download"
```
Comment (TomaszGaweda) [inline kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java:38]: Can you please use our local repository?

File: kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Override
+    protected void test(HazelcastInstance client, String name) throws MalformedURLException {
+
+        final int ITEM_COUNT = 10;
+
+        Properties connectorProperties = new Properties();
+        connectorProperties.setProperty("name", "datagen-connector");
+        connectorProperties.setProperty("connector.class", "io.confluent.kafka.connect.datagen.DatagenConnector");
+        connectorProperties.setProperty("tasks.max", "1");
```
Comment (TomaszGaweda) [inline kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java:57]: can we test with more tasks?

File: kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Override
+    protected void init(HazelcastInstance client) {
+    }
+
+    @Override
+    protected void test(HazelcastInstance client, String name) throws MalformedURLException {
+
+        final int ITEM_COUNT = 10;
```
Comment (TomaszGaweda) [inline kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java:52]: the test should run in a loop, see eg. MongoTest

**Review by TomaszGaweda — CHANGES_REQUESTED** [review]

**Review by AleksPeychev — APPROVED** [review]

**Review by olukas — CHANGES_REQUESTED** [review]
Adding my "Request changes" here to ensure this will not be merged until it will have proper quality.

Some of areas which has to be improved:
- if it's possible we should test in Jet soak tests with real Kafka Connect
- hazelcast-jet-ansible part of PR is missing (setup the test to run in Jet soak tests)
- since this is stream connector then the job should not finish but tun for the whole time of running testsuite, moreover we should have run it on both stable (due to memory leaks) and dynamic cluster (due to snapshotting)
- logging is totally missing - the test should run for 2days - we need to be able to see from client log that test is processing new data
- we should have some logic to check that test is processing some data (and fail if it stucked) - for example we should not be in the situation where we run 2 days soak tests and after 2 days we see the issue that pipeline is broken and did not processed any data from the start of the run

I haven't checked the test logic itself, just put here some comments about some Jet soak tests essentials which need to be done.

File: kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java (hunk 2 above)
Comment (orcunc) [reply kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java]: rev : 5686f9988ee346fec85eebc01e57a5fb3a096710. 
task.masks = 2

File: kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java (hunk 3 above)
Comment (orcunc) [reply kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java]: rev : 5686f9988ee346fec85eebc01e57a5fb3a096710
running tests in loop now

File: kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java (hunk 1 above)
Comment (orcunc) [reply kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java:40]: what do you mean local repository ?

Comment (orcunc) [reply kafka-connect-source/src/main/java/com/hazelcast/jet/tests/kafkaconnectsource/KafkaConnectSourceTest.java]: rev : 62052e37ab626ff865a3319e99368000b1956382

**Comment by TomaszGaweda** [comment]
From my PoV it's ok with the exception for comments given by Ondrej. Once those are addressed I can approve this PR :)

**Pull request closed without merging**